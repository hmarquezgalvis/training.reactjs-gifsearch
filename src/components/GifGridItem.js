import React from 'react';
import PropTypes from 'prop-types';

import './GifGridItem.css';

const GifGridItem = ({ title, url }) => {
  return (
    <>
      <div className="col-3">
        <img className="border border-white animate_animated animate__bounceIn"
          src={ url } 
          alt={ title } />
      </div>
    </>
  )
}
GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifGridItem;
