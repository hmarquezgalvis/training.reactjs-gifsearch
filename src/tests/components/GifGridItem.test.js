import { shallow } from 'enzyme';

import GifGridItem from '../../components/GifGridItem'

describe( 'Tests for <GifGridItem />', () => {

  test( 'must be show the right state of component', () => {

    const url = 'https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=46e040a5354ei2def6320v0iehtkqw7wxly2lkk09swh5fpu&rid=giphy.gif&ct=g';
    const title = 'One Punch Man GIF';

    const wrapper = shallow( <GifGridItem url={ url } title={ title } /> );
    expect( wrapper ).toMatchSnapshot();
  });

});