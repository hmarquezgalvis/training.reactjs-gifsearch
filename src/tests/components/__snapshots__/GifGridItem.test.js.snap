// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem /> must be show the right state of component 1`] = `
<Fragment>
  <div
    className="col-3"
  >
    <img
      alt="One Punch Man GIF"
      className="border border-white animate_animated animate__bounceIn"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=46e040a5354ei2def6320v0iehtkqw7wxly2lkk09swh5fpu&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;
